### ISSUE TITLE
Mising inport for Mongo retriever

### ISSUE THREAD

**Issue by estaar** [stanfordnlp/dspy#1375]
The `dspy/retrieve/mongodb_atlas_rm.py` is missing a `dspy` import. Running the import gives the following error.

`from dspy.retrieve.mongodb_atlas_rm import MongoDBAtlasRM`

```
Error: 
---> 71 class MongoDBAtlasRM(dspy.Retrieve):
     72     def __init__(
     73         self,
     74         db_name: str,
   (...)
     79         embedding_model: str = "text-embedding-ada-002",
     80     ):
     81         super().__init__(k=k)

NameError: name 'dspy' is not defined
```

**Comment by estaar** [comment]
This should consider using a `MONGO-CONNECTION-STRING` instead of asking for 
```
 self.username = os.getenv("ATLAS_USERNAME")
 self.password = os.getenv("ATLAS_PASSWORD")
 self.cluster_url = os.getenv("ATLAS_CLUSTER_URL")
   ```

**Comment by arnavsinghvi11** [comment]
Hi @estaar , fixed the import error #1400 . feel free to add the other MongoRM configuration changes via a PR!
